File: C/Lista_05(Matrizes)/Diagonal_Negativos.c
#include <stdio.h>

int main() {

    int n;

    printf("Qual a ordem da matriz? ");
    scanf("%d", &n);

    int mat[n][n];

    for(int i=0;i<n;i++){
        for(int j=0;j<n;j++){
            printf("Elemento [%d,%d]: ", i, j);
            scanf("%d", &mat[i][j]);
        }
    }

    printf("DIAGONAL PRINCIPAL:\n");

    for(int i=0;i<n;i++){
        printf("%d ", mat[i][i]);
    }

    int cont = 0;

    for(int i=0;i<n;i++){
        for(int j=0;j<n;j++){
            if(mat[i][j]<0){
                cont++;
            }
        }
    }

    printf("\nQUANTIDADE DE NEGATIVOS = %d", cont);

    return 0;

}

File: Java/Lista_05(Matrizes)/Diagonal_Negativos.java
import java.util.Scanner;

public class Diagonal_Negativos {

    public static void main(String[] args) {

        Scanner s = new Scanner(System.in);

        System.out.print("Qual a ordem da matriz? ");
        int n = s.nextInt();

        int mat[][] = new int[n][n];

        for (int i = 0; i < n; i++) {
            for (int j = 0; j < n; j++) {
                System.out.print("Elemento [" + i + "][" + j + "]: ");
                mat[i][j] = s.nextInt();
            }
        }

        System.out.println("DIAGONAL PRINCIPAL:");

        for (int i = 0; i < n; i++) {
            System.out.print(mat[i][i] + " ");
        }

        int contN = 0;

        for (int i = 0; i < n; i++) {
            for (int j = 0; j < n; j++) {
                if (mat[i][j] < 0) {
                    contN++;
                }
            }
        }

        System.out.print("\nQuantidade de negativos = " + contN);

        s.close();

    }
}